dashboard/lib/services/i18n/curriculum_sync_utils/sync_out.rb

```diff
@@ -68,6 +49,44 @@ def self.reorganize
           end
         end
 
+        # Perform any necessary postprocessing to the final translated strings,
+        # before writing them out to config files
+        def self.postprocess(result)
+          # We use URLs as keys for lessons in Crowdin, to make things easier for
+          # translators. For the actual translation files, though, we'd like to use
+          # more-standard keys.
+          if result.key?(:lessons)
+            rekeyed_lessons = result[:lessons].map do |lesson_url, lesson_data|
+              route_params = Rails.application.routes.recognize_path(lesson_url)
+              lesson = Lesson.joins(:script).
+                find_by(
+                  "scripts.name": route_params[:script_id],
+                  relative_position: route_params[:position].to_i,
+                  has_lesson_plan: true
+                )
+              unless lesson.present?
+                STDERR.puts "could not find lesson for url #{lesson_url.inspect}"
+                next
+              end
+              [Services::GloballyUniqueIdentifiers.build_lesson_key(lesson), lesson_data]
+            end
+            result[:lessons] = rekeyed_lessons.compact.to_h
+          end
+
+          # We also provide URLs to the translators for Resources only; because
+          # the sync has a side effect of applying Markdown formatting to
+          # everything it encounters, we want to make sure to un-Markdownify
+          # these URLs
```

Comment:
For my curiosity, what is an example of Markdownify an URL?

Reply:
our markdown processor automatically turns urls into [autolinks](http://sgmljs.net/docs/markdown-autolink-examples.html), so `https://example.com` will become `<https://example.com>`